diff --git a/cloudflare.install b/cloudflare.install
index ef23348..a1b9656 100644
--- a/cloudflare.install
+++ b/cloudflare.install
@@ -121,36 +121,3 @@ function cloudflare_update_10001() {

   return 'Cloudflare state data was successfully updated.';
 }
-
-/**
- * Convert zone IDs to ID=>name map.
- */
-function cloudflare_update_10002() {
-  $config = \Drupal::configFactory()->getEditable('cloudflare.settings');
-  if ($zone_ids = $config->get('zone_id')) {
-    $zones_used = [];
-    $zones_all = [];
-    $response = \Drupal::service('cloudflare.api_client')->listZones();
-    if ($response->success === TRUE) {
-      foreach ($response->result as $zone) {
-        $zones_all[$zone->id] = $zone->name;
-      }
-    }
-    else {
-      throw new UpdateException('Update 10002 failed with the following: ' . var_export($response->errors, TRUE));
-    }
-
-    foreach ($zone_ids as $zone_id) {
-      if (isset($zones_all[$zone_id])) {
-        $zones_used[$zone_id] = $zones_all[$zone_id];
-      }
-    }
-    $config
-      ->set('zones', $zones_used)
-      ->clear('zone_id')
-      ->clear('zone_name')
-      ->save();
-  }
-
-  return 'Cloudflare configuration ( cloudflare.settings ) was successfully updated.';
-}
